omg/entities/player.py

```diff
@@ -85,14 +107,26 @@ def update(self, mouse_x, mouse_y, delta_time):
         )
         self._regenerate_mana(delta_time)
 
-    def shoot(self):
-        """Shoots a projectile and informs the observes with an ProjectileShotEvent."""
-        if self.skills.current_size == 0 or self.current_mana < 20:
+        # Update combined elements
+        if len(self.to_be_combined_element_buffer) == 2:
+            # Get the name of the skill after combining elements
+            new_skill = "".join(self.to_be_combined_element_buffer[::])
+
+            # Update crafted skill slots
+            self._update_crafted_skill_slots(new_skill)
+
+            # Empty the element buffer
+            self.to_be_combined_element_buffer = []
+
+    def shoot(self, skill_name: str):
+        """Shoot a projectile and inform the observers."""
+        self.crafted_skill.set_skill_attributes(skill_name)
+        mana_cost = self.crafted_skill.mana_cost
+        if skill_name is None or self.current_mana < mana_cost:
```

Comment:
This line should be at the top. Also, you can use `if skill_name` , instead of `if skill_name is None` , since anything `falsy` would be allowed in this case (None is also an falsy value in python)

This line not being in the top also creates a bug, where just launching the game and pressing a skill button.